include/envoy/upstream/cluster_manager.h

```diff
@@ -149,12 +150,11 @@ class ClusterManagerFactory {
   /**
    * Allocate a cluster manager from configuration JSON.
    */
-  virtual ClusterManagerPtr clusterManagerFromJson(const Json::Object& config, Stats::Store& stats,
-                                                   ThreadLocal::Instance& tls,
-                                                   Runtime::Loader& runtime,
-                                                   Runtime::RandomGenerator& random,
-                                                   const LocalInfo::LocalInfo& local_info,
-                                                   AccessLog::AccessLogManager& log_manager) PURE;
+  virtual ClusterManagerPtr
+  clusterManagerFromJson(const Json::Object& config, const envoy::api::v2::Bootstrap& bootstrap,
```

Comment:
Any reason we are not just converting JSON to bootstrap like we are doing in other PRs? That seems like right long term approach?